target/linux/airoha/files/drivers/mfd/airoha-en7581-mfd.c

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: GPL-2.0+
+/*
+ * MFD driver for Airoha EN7581
+ */
+
+#include <linux/io.h>
+#include <linux/of.h>
+#include <linux/mfd/airoha-en7581-mfd.h>
+#include <linux/mfd/core.h>
+#include <linux/module.h>
+
+#include <linux/mfd/syscon.h>
+
+
+static const struct mfd_cell airoha_mfd_devs[] = {
+	{
+		.name = "pinctrl-airoha",
+	  	.of_compatible = "airoha,en7581-pinctrl",
+	},
+	{
+		.name = "airoha-pwm",
+	  	.of_compatible = "airoha,en7581-pwm",
+	},
+};
+
+static int airoha_mfd_probe(struct platform_device *pdev)
+{
+	struct device *dev = &pdev->dev;
+	struct airoha_mfd *mfd;
+	int ret;
+
+	mfd = devm_kzalloc(dev, sizeof(*mfd), GFP_KERNEL);
+	if (!mfd)
+		return -ENOMEM;
+	platform_set_drvdata(pdev, mfd);
+
+	mfd->base = devm_platform_ioremap_resource(pdev, 0);
+	if (IS_ERR(mfd->base))
+		return PTR_ERR(mfd->base);
+
+	mfd->chip_scu = syscon_regmap_lookup_by_phandle(dev->of_node,
```

Comment:
@Ansuel I guess we can use compatible string instead

Reply:
Main problem is that using absolute compatible is problematic for yaml schema as we don't have a way to reference them... IMHO phandle is better